src/main/java/roomescape/utility/TossPaymentClient.java

```diff
@@ -0,0 +1,72 @@
+package roomescape.utility;
+
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Map;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.web.client.RestClient;
+import org.springframework.web.client.RestClientException;
+import roomescape.dto.business.PaymentExceptionContent;
+import roomescape.dto.business.PaymentResult;
+import roomescape.exception.PaymentException;
+
+public class TossPaymentClient implements PaymentClient {
+
+    private static final String CONNECTION_ERROR_MESSAGE = "결제 서버에 연결이 실패하였습니다. 이 현상이 지속되는 경우 어드민에게 문의해주세요.";
+    private static final String CONFIRM_SERVER_FAIL_MESSAGE = "결제 연동 서버가 아파요. 관리자에게 문의해주세요.";
+
+    private final RestClient restClient;
+    private final String secretKey;
+    private final ObjectMapper statusParser = new ObjectMapper();
+    private final String paymentConfirmUrl;
+
+    public TossPaymentClient(
+            RestClient restClient,
+            String secretKey,
+            String paymentConfirmUrl
+    ) {
+        this.restClient = restClient;
+        this.secretKey = secretKey;
+        this.paymentConfirmUrl = paymentConfirmUrl;
+    }
+
+    @Override
+    public PaymentResult pay(String paymentKey, String orderId, long amount) {
+        Map<String, Object> requestBody = Map.of(
+                "paymentKey", paymentKey,
+                "orderId", orderId,
+                "amount", amount
+        );
+
+        try {
+            return doPay(requestBody);
+        } catch (RestClientException restClientException) {
+            throw new RestClientException(CONNECTION_ERROR_MESSAGE);
+        }
+    }
+
+    private PaymentResult doPay(Map<String, Object> requestBody) {
+        return restClient.post()
+                .uri(paymentConfirmUrl)
+                .body(requestBody)
+                .header("Authorization", createAuthHeaderConcise())
```

Comment:
이미 정의된 상수로 대체해볼 수 있을 것 같아요. 😃
 
```suggestion
                .header(HttpHeaders.AUTHORIZATION createAuthHeaderConcise())
```

Reply:
지금까지 HttpHeaders가 있는지 조차 모르고 있었습니다.
과거에 HttpStatus같은것은 사용했었는데, 왜 생각못했을까요 ㅠㅠ 알려주셔서 감사합니다 🙏

Reply:
[refactor: HttpHeader에서 정의된 상수 사용](https://github.com/woowacourse/spring-roomescape-payment/pull/192/commits/5d3d24863ef23b36807996c0f3ed27cffd65ee76)

Reply:
이런게 있다고 보통 생각을 못하니 한번도 안써보셨다면 모르는게 일반적이라고 생각해요. 😃
다음에는 Header의 key 값 설정할 때는 정의된 상수가 없는지 먼저 찾아보시지 않을까요?  🤗